core/services/relay/evm/ocr2keeper.go

```diff
@@ -124,11 +126,23 @@ func (r *ocr2keeperRelayer) NewOCR2KeeperProvider(rargs commontypes.RelayArgs, p
 	logProvider, logRecoverer := logprovider.New(r.lggr, client.LogPoller(), client.Client(), upkeepState, finalityDepth)
 	blockSub := evm.NewBlockSubscriber(client.HeadBroadcaster(), client.LogPoller(), finalityDepth, r.lggr)
 
+	kb, err := r.ocr2Keystore.Get(rargs.KeyBundleID)
+	if err != nil {
+		return nil, err
+	}
+
+	kr := evm.NewOnchainKeyringV3Wrapper(kb)
+
 	al := evm.NewActiveUpkeepList()
 	payloadBuilder := evm.NewPayloadBuilder(al, logRecoverer, r.lggr)
 
 	reg := evm.NewEvmRegistry(r.lggr, cfgWatcher.contractAddress, client,
-		registryContract, nil, al, logProvider, // TODO pass mercury credentials instead of nil
+		registryContract, &models.MercuryCredentials{
```

Comment:
We can also probably remove the `models.MercuryCredentials` type (which has been duplicated into common) and pass `rargs.MercuryCredentials` directly here?